engine/apps/alerts/migrations/0041_alertreceivechannel_unique_direct_paging_integration_per_team.py

```diff
@@ -0,0 +1,56 @@
+# Generated by Django 4.2.7 on 2023-11-28 10:45
+import logging
+
+from django.db import migrations, models
+from django.utils import timezone
+
+logger = logging.getLogger(__name__)
+
+
+def delete_duplicate_direct_paging_integrations(apps, schema_editor):
+    AlertReceiveChannel = apps.get_model("alerts", "AlertReceiveChannel")
+
+    # get (organization_id, team_id) pairs for teams that have more than one direct paging integration
+    duplicate_rows = AlertReceiveChannel.objects.values_list(
+        "organization_id", "team_id"
+    ).annotate(count=models.Count("id")).filter(integration="direct_paging", deleted_at__isnull=True, count__gt=1)
```

Comment:
Would it be better to filter before annotating (to avoid annotating over all integrations)? ie.
```
duplicate_rows = AlertReceiveChannel.objects.filter(
    integration="direct_paging", deleted_at__isnull=True
).values_list(
    "organization_id", "team_id"
).annotate(count=models.Count("id")).filter(count__gt=1)
```

Reply:
Both produce equivalent SQL queries, so I think the filter order doesn't really matter:

```mysql
SELECT organization_id, team_id, COUNT(id) AS count
FROM alerts_alertreceivechannel
WHERE (deleted_at IS NULL AND integration = 'direct_paging') 
GROUP BY organization_id, team_id
HAVING COUNT(id) > 1 
ORDER BY NULL
```